src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -558,15 +570,39 @@ private[fastq] object FastqDemultiplexer {
   /** Stores the minimal information for a single template read. */
   case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
                          sampleBarcode: Option[String], readNumber: Int, pairedEnd: Boolean, comment: Option[String],
-                         originalBases: Option[String] = None, originalQuals: Option[String] = None)
+                         originalBases: Option[String] = None, originalQuals: Option[String] = None) {
+
+    /** Masks bases that have a quality score less than or equal to a specified threshold.
+      * @param threshold The threshold for masking bases, exlusive. Bases with a quality score < threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: PhredScore, qualityEncoding: QualityEncoding): DemuxRecord = {
+      if (threshold <= 0 || this.quals.forall(_ >= threshold)) this else {
+        val bases = this.bases.toCharArray
+        for (i <- Range(0, this.quals.length)) if (quals.charAt(i) < threshold) bases(i) = 'N'
+        this.copy(bases = bases.mkString)
```

Comment:
This is still less efficient that it could be.  IIRC I suggested using `new String(bases)` instead of `bases.mkString`.  The former does a very efficient array copy, while the latter assembles the string one character at a time through a StringBuilder which is going to be much slower.

Reply:
Also the `forloop` construct in `FgbioDef` is more efficient that doing `for (i <- Range())`